python/cuml/test/test_linear_model.py

```diff
@@ -38,17 +39,25 @@
 from sklearn.model_selection import train_test_split
 
 
-def make_regression_dataset(datatype, nrows, ncols, n_info):
+def _make_regression_dataset_uncached(nrows, ncols, n_info):
     X, y = make_regression(
         n_samples=nrows, n_features=ncols, n_informative=n_info, random_state=0
     )
-    X = X.astype(datatype)
-    y = y.astype(datatype)
-    X_train, X_test, y_train, y_test = train_test_split(
-        X, y, train_size=0.8, random_state=10
-    )
+    return train_test_split(X, y, train_size=0.8, random_state=10)
 
-    return X_train, X_test, y_train, y_test
+
+@lru_cache(4)
+def _make_regression_dataset_from_cache(nrows, ncols, n_info):
+    return _make_regression_dataset_uncached(nrows, ncols, n_info)
+
+
+def make_regression_dataset(datatype, nrows, ncols, n_info):
+    if nrows * ncols < 1e8:  # Keep cache under 4 GB
```

Comment:
Why the choice of 4GB? That would (potentially) mean that 4GB cards _could_ have issues in the future, no? Just wondering the rationale of why 4 and whether it is the best option indeed

Reply:
Yep, this is somewhat arbitrary, but it also has a fair margin of safety built in. If we had four consecutive uncached calls that all required exactly `1e8 - 1` float64s, we would use 3.2 GB for this cache. Practically speaking, we stay far below this theoretical maximum.

That being said, I _am_ a bit uncomfortable with  the possibility that we do this once here and then create another 3.2 GB cache somewhere else and so on. I'll see if there's some simple way to flush the cache once we've finished with this set of tests.

Reply:
I think reviewing this can be part of the issue #3080 you created indeed

Reply:
Great, thanks! Just explicitly called this out in the issue.